adept/adroit.py

```diff
@@ -0,0 +1,102 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import equinox as eqx
+import jax
+import jax.numpy as jnp
+import mlflow
+import numpy as np
+
+
+class MLRunId(eqx.Module):
+    """
+    An mlflow run id encoded as a JAX array of bytes.
+    This class is needed to pass around the run id in JIT-compiled sections.
+    """
+
+    byte_array: jax.Array
+
+    def __init__(self, id: str):
+        self.byte_array = jnp.frombuffer(id.encode("UTF-8"), dtype=jnp.int8)
+
+    def __str__(self):
+        return bytes(np.array(self.byte_array)).decode("UTF-8")
+
+    @staticmethod
+    def example():
+        return MLRunId("0" * 32)
+
+
+class AdroitModule(eqx.Module, ABC):
+    with_mlflow: bool
+
+    @staticmethod
+    def create_mlflow_run_in_callback(dummy_arg):
+        def create_mlflow_run(dummy_arg):
+            with mlflow.start_run() as run:
+                return MLRunId(run.info.run_id)
+
+        return jax.experimental.io_callback(
+            create_mlflow_run, MLRunId.example(), dummy_arg, sharding=None, ordered=False
```

Comment:
```suggestion
            create_mlflow_run, MLRunId.example(), dummy_arg,
```

Don't see why we want to explicitly provide the default kwargs? Or is this for documentation?